dev/fattest.simplicity/src/componenttest/topology/utils/tck/TCKUtilities.java

```diff
@@ -704,7 +723,27 @@ static String getArtifactoryUser() {
      * @return the token, or {@code null} if none is configured
      */
     static String getArtifactoryToken() {
-        return System.getProperty(FAT_TEST_PREFIX + ARTIFACTORY_TOKEN_KEY);
+        return validateStringProperty(System.getProperty(FAT_TEST_PREFIX + ARTIFACTORY_TOKEN_KEY));
+    }
+
+    /**
+     * This validates the result string from using System.getProperty().
+     * See launch.xml line 322 we first check if a property was set via a command line property
+     * If not we assume the property was set as an environment property.
+     *
+     * Which mean fat.test.artifactory.download.server could equal "${env.artifactory.download.server}"
+     *
+     * @param  result the string to validate
+     * @return        null if the string was invalid, itself otherwise.
+     */
+    static String validateStringProperty(String result) {
+        if (result == null)
+            return result;
+
+        if (result.startsWith("${"))
+            return null;
+
+        return result;
```

Comment:
Validating usually means "check it and throw an exception if it's wrong", it seems weird that a method named "validate" is making changes to the data.

Suggest renaming it to something like `normalizeStringProperty` or `fixupStringProperty`

Reply:
It might make sense for this method to also convert an empty string to `null`.